lightning/src/routing/network_graph.rs

```diff
@@ -1658,4 +1747,132 @@ mod tests {
+		let node_2_privkey = &SecretKey::from_slice(&[41; 32]).unwrap();
+		let node_id_1 = PublicKey::from_secret_key(&secp_ctx, node_1_privkey);
+		let node_id_2 = PublicKey::from_secret_key(&secp_ctx, node_2_privkey);
+		let node_1_btckey = &SecretKey::from_slice(&[40; 32]).unwrap();
+		let node_2_btckey = &SecretKey::from_slice(&[39; 32]).unwrap();
+
+		let short_channel_id = 0;
+		let chain_hash = genesis_block(Network::Testnet).header.bitcoin_hash();
+
+		{
+			// There is no nodes in the table at the beginning.
+			let network = net_graph_msg_handler.network_graph.read().unwrap();
+			assert_eq!(network.get_nodes().len(), 0);
+		}
+
+		{
+			// Announce a channel we will update
+			let unsigned_announcement = UnsignedChannelAnnouncement {
+				features: ChannelFeatures::empty(),
+				chain_hash,
+				short_channel_id,
+				node_id_1,
+				node_id_2,
+				bitcoin_key_1: PublicKey::from_secret_key(&secp_ctx, node_1_btckey),
+				bitcoin_key_2: PublicKey::from_secret_key(&secp_ctx, node_2_btckey),
+				excess_data: Vec::new(),
+			};
+
+			let msghash = hash_to_message!(&Sha256dHash::hash(&unsigned_announcement.encode()[..])[..]);
+			let valid_channel_announcement = ChannelAnnouncement {
+				node_signature_1: secp_ctx.sign(&msghash, node_1_privkey),
+				node_signature_2: secp_ctx.sign(&msghash, node_2_privkey),
+				bitcoin_signature_1: secp_ctx.sign(&msghash, node_1_btckey),
+				bitcoin_signature_2: secp_ctx.sign(&msghash, node_2_btckey),
+				contents: unsigned_announcement.clone(),
+			};
+			match net_graph_msg_handler.handle_channel_announcement(&valid_channel_announcement) {
+				Ok(_) => (),
+				Err(_) => panic!()
+			};
+		}
+
+		{
+			// At the start, the node_1 <-> node_2 channel's DirectionalChannelInfos should be 0.
+			let mut network = net_graph_msg_handler.network_graph.write().unwrap();
+			let chan_id = network.get_nodes().get(&node_id_1).unwrap().channels[0];
+			let channel_info = network.channels.get_mut(&chan_id).unwrap();
+			// Assign a DirectionalChannelInfo and ChannelScore object to one_to_two of
+			// channel_info
+			let chan_score = ChannelScore {
+				payment_sent_score: 0,
+				payment_failed_score: 0,
+			};
+			let dummy_routing_fee = RoutingFees {
+				base_msat: 0,
+				proportional_millionths: 0,
+			};
+			let dir_chan_info = DirectionalChannelInfo {
+				last_update: 0,
+				enabled: true,
+				cltv_expiry_delta: 0,
+				htlc_minimum_msat: 0,
+				fees: dummy_routing_fee,
+				last_update_message: None,
+				channel_score: chan_score,
+			};
+			channel_info.one_to_two = Some(dir_chan_info);
+			let dir_one_to_two = channel_info.one_to_two.as_mut().unwrap();
+			assert_eq!(dir_one_to_two.channel_score.payment_sent_score, 0);
+			assert_eq!(dir_one_to_two.channel_score.payment_failed_score, 0);
+		}
+
+		{
+			// Assume a payment is made, and node 1 is learning of a PaymentSent event. It now calls
+			// its score_payment_sent_for_route on itself.
+			let route_hop = RouteHop {
+								pubkey: node_id_1,
```

Comment:
Is indentation intentional ? It's maybe better for clarity, but you can also add whitespace before/after `route_hop` declaration.

Reply:
Oops, no, not intentional. This is fixed now!